jwst/resample/resample.py

```diff
@@ -335,13 +353,20 @@ def resample_many_to_many(self, input_models):
                     output_name = os.path.join(self.output_dir, output_name)
                 output_model.save(output_name)
                 log.info(f"Saved model in {output_name}")
-                self.output_models.append(output_name)
+                output_models.append(output_name)
             else:
-                self.output_models.append(output_model.copy())
+                output_models.append(output_model.copy())
             output_model.data *= 0.
             output_model.wht *= 0.
 
-        return self.output_models
+        if not self.in_memory:
+            # build ModelLibrary as an association from the output files
+            # this saves memory if there are multiple groups
+            asn = asn_from_list(output_models, product_name='outlier_i2d')
+            asn_dict = json.loads(asn.dump()[1]) # serializes the asn and converts to dict
+            return ModelLibrary(asn_dict, on_disk=True)
```

Comment:
@melanieclarke @braingram Melanie started a conversation on Slack related to [this bug ticket](https://jira.stsci.edu/browse/JP-3717) asking whether it was still necessary to save the drizzled models in this way with ModelLibrary being implemented.  These files are deleted later by [this line](https://github.com/spacetelescope/jwst/blob/582dbd953ac9dcba1b353dfcc26646913ccfeecb/jwst/outlier_detection/imaging.py#L110) in outlier detection if `save_intermediate_results` is False.

Brett, if I understand ModelLibrary correctly, it is not possible to add these models to the library one at a time, is that right?  I think Melanie was curious to hear your thoughts on why that design decision was made; this appears to be one use-case where that functionality might be helpful, although I know there are other "bad" patterns that are avoided because of the design.

I'm also wondering if I have the mechanics of this right: is it true that the ModelLibrary, after reading the association that gets created here, will only write temp files if `on_disk=True` _and_ a model is borrowed and then shelved with `modify=True`?  Otherwise it will read it from the original file?

Reply:
A temporary directory makes sense.  I suppose the justification for numpy arrays instead of datamodels would be that datamodels need to be validated every time they are loaded, and they might get loaded many times by `get_sections` if the total data volume is large?

At this point I'm biased toward calling everything "beyond the scope of this PR" so it can get merged, lol.  However I still think this is beyond the scope, because the PR basically retains the old behavior - these files are also written on main.  I'm willing to take on the bug ticket though if that's helpful.

Reply:
I agree, beyond the scope for this PR!  But we should discuss temporary files as a possible solution for JP-3717.  

And thanks for the clarification. I suspect allowing ModelLibrary to be appended to would be useful for future development, but it can wait until there is a real use case to justify the additional complexity.